Devource-co/kencime-backend
# #168309377 Users should be able to signup

#### What does this PR do?
Users should be able to signup to the application and also verify their phone number or emails

#### Description of Task to be completed?
- signup new users
- verify their email
- resend the verification code

#### How should this be manually tested?
- git clone the repo
- run `npm i`
- follow readme to setup the db
- run `npm run watch`
- go to the URL provided `http://localhost:4000/graphql`
- run on the playground, mutation 
```graphql
mutation {
  register(
    data: {
      name: "bonvic"
      useEmail: false
      email: ""
      phone: "+1(240)428-6668"
      password: "gqe536rf3r"
    }
  ) {
    id
    name
    phone
    email
  }
}
```
To signup new user

- run mutation, to verify account
```graphql
mutation {
  verifyAccount(data: { id: "<id of the user from signup>", verificationCode: "<sent code>"}){
   id
  }
}
```

run mutation, to resend the code:
```graphql
mutation {
  resendVerification(data: { useEmail: false, email: "", phone: "<number of the user>"}) {
    id
    verificationCode
  }
}
```
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?

[#168309377](https://www.pivotaltracker.com/n/projects/2396202/stories/168309377)

#### Screenshots

![image](https://user-images.githubusercontent.com/46062609/78668701-68d23900-78db-11ea-9cb2-823ba2050a45.png)
